src/free.rs

```diff
@@ -35,6 +35,41 @@ pub use crate::merge_join::merge_join_by;
 #[cfg(feature = "use_alloc")]
 pub use crate::rciter_impl::rciter;
 
+/// Iterate `iterable` with a particular value inserted between each element.
+///
+/// [`IntoIterator`] enabled version of [`Iterator::intersperse`].
+///
+/// ```
+/// use itertools::intersperse;
+///
+/// itertools::assert_equal(intersperse((0..3), 8), vec![0, 8, 1, 8, 2]);
+/// ```
+pub fn intersperse<I>(iterable: I, element: I::Item) -> Intersperse<I::IntoIter>
+    where I: IntoIterator,
+          <I as IntoIterator>::Item: Clone
+{
+    Itertools::intersperse(iterable.into_iter(), element)
```

Comment:
I think we canonically use `iterable.into_iter().intersperse(element)`. Could you adapt this?

Reply:
I think this will eventually break because of https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.intersperse . This is actually the motivation for providing the free functions - so there's no conflict when the core `intersperse` and `intersperse_with` are stable.